### DISCUSSION TITLE
Add emoji reactions to posts, comments.

### DISCUSSION THREAD

**Discussion by dessalines** [LemmyNet/lemmy-ui#1502] [General]

**Comment by Nutomic** [comment]
You really think this is a good idea?

**Comment by StaticallyTypedRice** [comment]
Personally, I'd leave this out. It might work for Mastodon because posts tend to be very short, but the thing that sets Lemmy apart from Mastodon is that this platform is much more geared to discussion and not reaction. Putting a smiley face doesn't convey much more information than simply upvoting it and IMO would clutter up the UI.

**Comment by Nutomic** [comment]
Mastodon doesnt have reactions, only favs which work differently. Not even Reddit has this, so I really dont think that this is going to work.

**Comment by StaticallyTypedRice** [comment]
Another example would be when Facebook implemented reactions and almost everyone hated it (though to be fair it's was also because it was seen as a cop out from implementing a true dislike button).

**Comment by dessalines** [comment]
That poll showed that most ppl do want reactions, and most communication platforms have them nowadays: matrix, discord, github, fb. They're used super-frequently in chat matrixes and discords that I'm in, and ppl showed they do want them here too. Twitter doesn't have them because it would be another way for people to downvote the blue checkmarks.

Don't worry, if i implement them i'll have a user setting to hide them similar to the avatars.

**Comment by Nutomic** [comment]
Sure but none of these platforms are similar to Lemmy. And just because some people in a poll voted for it doesn't mean its necessarily a good idea. Besides, I think we have a lot of issues that are more important.

**Comment by dessalines** [comment]
Pleroma just added this now too :wink:

**Comment by n1trux** [comment]
It's two years later, do you still have a lot of issues that are more important?

Lemmy is similar to a lot of other platforms in some ways, e.g. that there's always a "starter post" with a title and no loosely connected graph of comments or posts. This is similar to both traditional forums (Flarum and Discourse have emoji reactions as addons) and Facebook (without the title, but Facebook has a singular top comment – and Emoji reactions). It is also similar to Matrix in that Lemmy has communities (which are like rooms, or rather… Matrix communities, which consist of a list of rooms, basically).

And it's also similar to messaging applications in that you can send messages to other users in a kind of "open group chat" and react to them with emoji.

IMHO Emoji could also be used to influence the positioning of posts, so a post with lots of happy emoji and lots of upvotes at the same time might be positioned higher than a post with the same amount of upvotes, but more angry faces.

**Comment by dessalines** [comment]
Yes, we have many other issues that are really important. There are currently only two devs working on lemmy. 

> IMHO Emoji could also be used to influence the positioning of posts, so a post with lots of happy emoji and lots of upvotes at the same time might be positioned higher than a post with the same amount of upvotes, but more angry faces.

I'm very much against this; lemmy already has votes for ranking / sorting, and picture emojis aren't comparable in any way.

**Comment by dullbananas** [comment]
If this is implemented, then it should be an opt-in feature that lets moderators add reaction types to a community

**Comment by L3v3L** [comment]
This might get super confusing for normal users. People might think adding a heart to a post is better than adding a upvote for example. Then you have to do the hard task of educating the users that emotes don't have any play in ranking. Fun idea though.

**Comment by alectrocute** [comment]
Not a fan of this one.

**Comment by RustoMCSpit** [comment]
this feels like what reddit did with its reactions, it could be cool if you coukd see 'reactions' but onky from peiole uou followed so every now and then youd see your friend put a cool emoji on a comment

**Reply by techno156** [reply]
Although it seems like it would be ripe for spam. For reactions, you already have the comments and the voting system to take care of things.

**Reply by RustoMCSpit** [reply]
not really, if we limit it to only friends being able to react and see what other friends reacted? maybe have it public but emojis expire after ten minutes??
